### PR TITLE
Add original role to the role mapping list in tenant.conf

### PR DESCRIPTION
**Pull request by chashikajw**
As $subject this will append the original role also to the role mapping list in the tenant.conf.

Fixes wso2/product-apim#9152

### CODE REVIEW

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -11614,6 +11614,25 @@ public static void updateTenantConfRoleAliasMapping(JSONObject newRoleMappingJso
...
         }
+
+        //append original role to the role mapping list
+        Set<Map.Entry<String, JsonElement>> roleMappingEntries = newRoleMappingJson.entrySet();
+        for (Map.Entry<String, JsonElement> entry: roleMappingEntries) {
+            List<String> currentRoles = Arrays.asList(String.valueOf(entry.getValue()).split(","));
+            boolean isOriginalRoleAlreadyInroles = false;
+            for (String role: currentRoles) {
+                if (role.equals(entry.getKey())) {
+                    isOriginalRoleAlreadyInroles = true;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:11625]:
```suggestion
                    isOriginalRoleAlreadyInRoles = true;
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 2]
```diff
@@ -11614,6 +11614,25 @@ public static void updateTenantConfRoleAliasMapping(JSONObject newRoleMappingJso
...
+                    break;
+                }
+            }
+
+            if (!isOriginalRoleAlreadyInroles) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:11630]:
```suggestion
            if (!isOriginalRoleAlreadyInRoles) {
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -11614,6 +11614,25 @@ public static void updateTenantConfRoleAliasMapping(JSONObject newRoleMappingJso
         } catch (UserStoreException e) {
             APIUtil.handleException("Couldn't read tenant configuration from User Store", e);
         }
+
+        //append original role to the role mapping list
+        Set<Map.Entry<String, JsonElement>> roleMappingEntries = newRoleMappingJson.entrySet();
+        for (Map.Entry<String, JsonElement> entry: roleMappingEntries) {
+            List<String> currentRoles = Arrays.asList(String.valueOf(entry.getValue()).split(","));
+            boolean isOriginalRoleAlreadyInroles = false;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:11622]:
```suggestion
            boolean isOriginalRoleAlreadyInRoles = false;
```

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 4]
```diff
@@ -11752,6 +11771,7 @@ public static boolean isCrossTenantSubscriptionsEnabled() {
         return false;
     }
 
+
```
Comment (kavishkafernando) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:11774]: Shall we remove this extra line?

Comment (chashikajw) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:11774]: Fixed in 3e4283c7d975e6e9d7c32feac5a0f5f1914359ed

**Pull request merged**